### ISSUE TITLE
[Inductor] argmax/max returns incorrect indices for boolean tensors on CUDA

### ISSUE THREAD

**Issue by Aminsed** [pytorch/pytorch#174069]
## Describe the bug

When using `torch.compile` with the Inductor backend on CUDA, `max(dim)[1]` (argmax) returns incorrect indices for boolean tensors. The eager backend returns correct results.

## Minimal Reproduction

```python
import torch
import torch._dynamo

device = 'cuda'

def fn(x):
    return x.max(1, keepdim=True)[1]

# Boolean tensor with True starting at position 4
mask = torch.zeros(4, 8, 32, device=device, dtype=torch.bool)
mask[:, 4:, :] = True

result_eager = fn(mask)
torch._dynamo.reset()
result_compiled = torch.compile(fn, backend='inductor')(mask)

print(f'Expected index: 4')
print(f'Eager result: {result_eager[0, 0, 0].item()}')
print(f'Inductor result: {result_compiled[0, 0, 0].item()}')
# Output:
# Expected index: 4
# Eager result: 4
# Inductor result: 1  # WRONG!
```

## Results

| Tensor dtype | Eager | Inductor | Correct? |
|--------------|-------|----------|----------|
| bool | 4 | 1 | No |
| float | 4 | 4 | Yes |
| int | 4 | 4 | Yes |

The bug only affects boolean tensors.

## Impact

This bug affects any code that uses `argmax` or `max(...)[1]` on boolean tensors when compiled with Inductor on CUDA. For example, it can cause incorrect behavior in `cumprod_backward` when using composite-compliant implementations with `at::where`.

## Versions

- PyTorch: 2.11.0a0 (main branch, commit a3bda4952b3)
- CUDA: 12.5

cc @ezyang @gchanan @kadeng @msaroufim @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @muchulee8 @amjames @aakhundov @coconutruben @jataylo @isuruf
